src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertySymbol.cs

```diff
@@ -1285,7 +1285,8 @@ private ImmutableArray<ParameterSymbol> ComputeParameters(Binder binder, BasePro
                 }
                 else if ((object)_setMethod != null && param.Name == ParameterSymbol.ValueParameterName)
                 {
-                    diagnostics.Add(ErrorCode.ERR_DuplicateGeneratedName, _location, param.Name);
+                    // note: there can only ever be one location for param, as it cannot be used in partial for a property
+                    diagnostics.Add(ErrorCode.ERR_DuplicateGeneratedName, param.Locations[0], param.Name);
```

Comment:
No, FirstOrDefault alone won't be fine, because we don't want default under any circumstances. _location is something that was already used here, so falling back to it is safe for this change (we are not introducing a **new** break/problem). Besides, judging by implementation of Locations property in this class, I find it very unlikely that _location could be null. Having an array with null in it is much-much worse and affects way more than returning an empty array, which case we are trying to mitigate.